Sources/ForageSDK/Foundation/Network/Collector.swift

```diff
@@ -323,6 +337,169 @@ class BasisTheoryWrapper: VaultCollector {
     }
 }
 
+// Wrapper class for Forage internal vault
+class ForageVaultWrapper: VaultCollector {
+    var customHeaders: [String: String] = [:]
+    let textElement: UITextField
+    
+    private let forageVaultConfig: ForageVaultConfig
+    private let logger: ForageLogger?
+    private let session: URLSessionProtocol
+    
+    init(
+        textElement: UITextField,
+        forageVaultConfig: ForageVaultConfig,
+        logger: ForageLogger? = DatadogLogger(ForageLoggerConfig(prefix: "Rosetta")),
+        // Taking session as an injected dependency to make this easier to test
+        session: URLSessionProtocol = URLSession.shared
+    ) {
+        self.textElement = textElement
+        self.customHeaders = [:]
+        self.forageVaultConfig = forageVaultConfig
+        self.logger = logger
+        self.session = session
+    }
+    
+    func setCustomHeaders(headers: [String : String], xKey: [String : String]) {
+        customHeaders = headers
+    }
+    
+    func sendData<T: Decodable>(path: String, vaultAction: VaultAction, extraData: [String : Any], completion: @escaping (T?, ForageError?) -> Void) {
+        var body = [String: String]()
+
+        // grab the payment method token and add it to the request body
+        for (key, value) in extraData {
+            if key == tokenKey, let paymentMethodToken = value as? String {
+                do {
+                    let token = try getPaymentMethodToken(paymentMethodToken: paymentMethodToken)
+                    body[key] = token
+                } catch {
+                    logger?.critical(
+                        "Failed to send data to Rosetta proxy. Rosetta token not found on card",
+                        error: error,
+                        attributes: nil
+                    )
+                    return completion(nil, CommonErrors.UNKNOWN_SERVER_ERROR)
+                }
+            }
+        }
```

Comment:
can these be a separate method or moved to getPaymentMethodToken? so that `sendData` is leaner

```swift
func getRosettaTokenOrNil() -> String? {
 for (key, value) ...
}

guard let token = getRosettaTokenOrNull() else {
  return completion(nil, CommonErrors.UNKNOWN_SERVER_ERROR)
}

Reply:
[092d569](https://github.com/teamforage/forage-ios-sdk/pull/169/commits/092d569faa93566a19717da88563a0ed732313fe)